### PR TITLE
feat: add emotes support

### PR DESCRIPTION
**Pull request by lmssiehdev**
gib feedback

### CODE REVIEW

File: shared/net.js [hunk 1]
```diff
@@ -1140,7 +1155,7 @@ class EmoteMsg {
         this.pos = s.readVec(0, 0, 1024, 1024, 16);
         this.type = s.readGameType();
         this.isPing = s.readBoolean();
-        s.readBits(6);
```
Comment (lmssiehdev) [inline shared/net.js:1143]: couldn't read 6 bits

File: server/src/game.ts [hunk 2]
```diff
@@ -107,8 +123,17 @@ export class Game {
...
+                for (const emote of this.emotes) {
+                    if (emote.playerId === player.id || !emote.isPing) {
+                        player.emotes.add(emote);
+                    }
+                }
+            }
+
             player.sendMsgs();
         }
+        if (this.emotes.size) this.emotes = new Set<Emote>();
```
Comment (leia-uwu) [inline server/src/game.ts]: just do `this.emotes.clear()`

File: shared/net.js [hunk 3]
```diff
@@ -904,7 +908,13 @@ class JoinMsg {
         this.proxy = s.readBoolean();
         this.otherProxy = s.readBoolean();
         this.bot = s.readBoolean();
-        s.readAlignToNextByte();
```
Comment (leia-uwu) [inline shared/net.js:907]: should keep the byte alignment

File: shared/net.js (hunk 1 above)
Comment (leia-uwu) [reply shared/net.js:1143]: because its 5 not 6, oops, my fault

**Review by leia-uwu — CHANGES_REQUESTED** [review]

File: shared/net.js (hunk 3 above)
Comment (lmssiehdev) [reply shared/net.js:907]: oops, didn't intend to delete it.

**Pull request merged**